src/ir_Coolix.cpp

```diff
@@ -98,112 +98,172 @@ void IRsend::sendCOOLIX(uint64_t data, uint16_t nbits, uint16_t repeat) {
 //   https://github.com/crankyoldgit/IRremoteESP8266/issues/484
 IRCoolixAC::IRCoolixAC(const uint16_t pin, const bool inverted,
                        const bool use_modulation)
-    : _irsend(pin, inverted, use_modulation) { stateReset(); }
-
-void IRCoolixAC::stateReset() { setRaw(kCoolixDefaultState); }
+    : _irsend(pin, inverted, use_modulation),
+      coolixState(IRCoolixAC::sOff),
+      fanSpeed(kCoolixFanAuto),
+      acTemperature(kCoolixTempMin),
+      acSensorTemperature(kCoolixSensorTempMin),
+      acMode(kCoolixCool),
+      turboFlag(false),
+      ledFlag(false),
+      cleanFlag(false),
+      sleepFlag(false),
+      zoneFollowFlag(false),
+      swingCounter(0),
+      swingHCounter(0),
+      swingVCounter(0) { cmdReset(); }
+
+void IRCoolixAC::cmdReset() { setRaw(kCoolixDefaultState); }
 
 void IRCoolixAC::begin() { _irsend.begin(); }
 
 #if SEND_COOLIX
 void IRCoolixAC::send(const uint16_t repeat) {
-  _irsend.sendCOOLIX(remote_state, kCoolixBits, repeat);
+  _irsend.sendCOOLIX(remote, kCoolixBits, repeat);
 }
 #endif  // SEND_COOLIX
 
-uint32_t IRCoolixAC::getRaw() { return remote_state; }
+void IRCoolixAC::send(uint64_t data, uint16_t nbits, const uint16_t repeat) {
+  _irsend.sendCOOLIX(data, nbits, repeat);
+}
+
+uint32_t IRCoolixAC::getRaw() { return remote; }
 
 void IRCoolixAC::setRaw(const uint32_t new_code) {
-  remote_state = new_code;
-  saved_state = new_code;
+  if(!handleSpecialState(new_code)){
+    // it isn`t special so might afect Temp|mode|Fan
+    if( new_code == kCoolixCmdFan ){
+      setMode(kCoolixFan);
+    }
+    else {
+      // must be a command changing Temp|Mode|Fan
+      // it is safe to just copy to remote var
+      remote = new_code;
+      setMode((new_code & kCoolixModeMask)>>2);
+      setFan((new_code & kCoolixFanMask)>>13);
```

Comment:
You should use `getMode()` and `getFan()`, and do the bit math there.

See comment on `getFan()`. Same goes for `getMode()` too.

Reply:
It is supposed to be true if we are still keeping state on "remote" attribute, otherwise getMode and getFan shall be returning the _mode or _fan attributes.